File: barq-sdk-go/async_smoke_test.go
package barq

import (
	"context"
	"os"
	"strings"
	"testing"
)

func TestAsyncInsertReturnsRequestID(t *testing.T) {
	client := NewClient(Config{
		BaseURL: getenv("BARQ_BASE_URL", "http://127.0.0.1:8080"),
		APIKey:  os.Getenv("BARQ_API_KEY"),
	})
	collection := getenv("BARQ_TEST_COLLECTION", "sdk-go-async")

	ctx := context.Background()

	if err := client.CreateCollection(ctx, CreateCollectionRequest{
		Name:      collection,
		Dimension: 2,
		Metric:    "Cosine",
	}); err != nil {
		t.Fatalf("create collection: %v", err)
	}

	requestID, err := client.InsertAsync(ctx, collection, InsertRequest{
		ID:      "go-async-doc",
		Vector:  []float32{1.0, 0.0},
		Payload: []byte(`{"sdk":"go","mode":"async"}`),
	})
	if err != nil {
		t.Fatalf("async insert: %v", err)
	}

	if !strings.HasPrefix(requestID, "ingest-") {
		t.Fatalf("unexpected request id %q", requestID)
	}
}

File: barq-sdk-ts/test/async_smoke.test.js
const test = require("node:test");
const assert = require("node:assert/strict");
const { BarqClient } = require("../dist/index.js");

test("typescript async insert returns request id", async () => {
    const client = new BarqClient({
        baseUrl: process.env.BARQ_BASE_URL || "http://127.0.0.1:8080",
        apiKey: process.env.BARQ_API_KEY ?? "",
    });
    const collection = process.env.BARQ_TEST_COLLECTION;

    assert.ok(collection, "BARQ_TEST_COLLECTION must be set");
    await client.createCollection({
        name: collection,
        dimension: 2,
        metric: "Cosine",
    });

    const requestId = await client
        .collection(collection)
        .insertAsync("ts-async-doc", [1.0, 0.0], { sdk: "ts", mode: "async" });

    assert.match(requestId, /^ingest-/);
});
